test/e2e/common/node_lease.go

```diff
@@ -68,7 +69,58 @@ var _ = framework.KubeDescribe("[Feature:NodeLease][NodeAlphaFeature:NodeLease]"
 				}
 				return nil
 			}, time.Duration(*lease.Spec.LeaseDurationSeconds)*time.Second,
-				time.Duration(*lease.Spec.LeaseDurationSeconds/3)*time.Second)
+				time.Duration(*lease.Spec.LeaseDurationSeconds/4)*time.Second)
+		})
+
+		It("the kubelet should report node status infrequently", func() {
+			nodeClient := f.ClientSet.CoreV1().Nodes()
+			var (
+				err            error
+				node           *corev1.Node
+				readyCondition *corev1.NodeCondition
+			)
+
+			By("wait until node is ready")
```

Comment:
I think we have some existing utility functions written for this. Maybe try searching something for `WaitForNodeToBeReady`...

Reply:
The util function is handy. But here we need to get the heartbeat time in order to compare with the next heartbeat time, and check the heartbeat time difference.